src/main/java/com/google/cloud/tools/appengine/api/deploy/DeployConfiguration.java

```diff
@@ -16,32 +16,68 @@
 
 package com.google.cloud.tools.appengine.api.deploy;
 
-import com.google.cloud.tools.appengine.api.Configuration;
+import com.google.auto.value.AutoValue;
 import java.io.File;
 import java.util.List;
 import javax.annotation.Nullable;
 
 /** Configuration for {@link AppEngineDeployment#deploy(DeployConfiguration)}. */
-public interface DeployConfiguration extends Configuration {
+@AutoValue
+public abstract class DeployConfiguration {
+  /** GCS storage bucket used for staging files associated with deployment. */
+  @Nullable
+  public abstract String getBucket();
 
+  /** List of deployable target directories or yaml files. */
   @Nullable
-  List<File> getDeployables();
+  public abstract List<File> getDeployables();
 
+  /** Docker image to use during deployment (only for app.yaml deployments). */
   @Nullable
-  String getBucket();
+  public abstract String getImageUrl();
 
+  /** Google Cloud Project ID to deploy to. */
   @Nullable
-  String getImageUrl();
+  public abstract String getProjectId();
 
+  /** Promote the deployed version to receive all traffic. */
   @Nullable
-  Boolean getPromote();
+  public abstract Boolean getPromote();
 
+  /** The App Engine server to use. Users typically will never set this value. */
   @Nullable
-  String getServer();
+  public abstract String getServer();
 
+  /** Stop the previous running version when deploying and promoting a new version. */
   @Nullable
-  Boolean getStopPreviousVersion();
+  public abstract Boolean getStopPreviousVersion();
 
+  /** Version to deploy. */
   @Nullable
-  String getVersion();
+  public abstract String getVersion();
+
+  public static Builder builder() {
+    return new AutoValue_DeployConfiguration.Builder();
+  }
+
+  @AutoValue.Builder
+  public abstract static class Builder {
```

Comment:
Why a builder?

Reply:
Some of these config classes have a whole lot of optional parameters, and I'd like them to all be builders for consistency.